apps/artemis-web/pages/adminpanel.tsx

```diff
@@ -2,13 +2,28 @@ import Head from 'next/head';
 import React from 'react';
 import { useJWT } from '../hooks/useJWT';
 import { useRouter } from 'next/router';
+import DefaultErrorPage from 'next/error';
 
 const AdminPanelPage = (props) => {
   const [user, loading] = useJWT();
 
   const router = useRouter();
   if (!user && !loading) router.push('signin');
 
+  if (user && user.role !== 'admin') {
+    return (
```

Comment:
try to move this part to an external function to remove duplication

Reply:
I added a hoc to solve this.